plugin/helpers_bots_test.go

```diff
@@ -153,3 +153,179 @@ func TestEnsureBot(t *testing.T) {
 		})
 	})
 }
+
+func TestKVGetJSON(t *testing.T) {
+	setupAPI := func() *plugintest.API {
+		return &plugintest.API{}
+	}
+
+	t.Run("KVGet error", func(t *testing.T) {
+		p := &plugin.HelpersImpl{}
+
+		api := setupAPI()
+		api.On("KVGet", plugin.BOT_USER_KEY).Return(nil, &model.AppError{})
+		p.API = api
+
+		var dat map[string]interface{}
+
+		err := p.KVGetJSON(plugin.BOT_USER_KEY, dat)
+
+		api.AssertExpectations(t)
+		assert.NotNil(t, err)
+		assert.Nil(t, dat)
+	})
+
+	t.Run("Malformed JSON", func(t *testing.T) {
+		key := "test-key"
+
+		p := &plugin.HelpersImpl{}
+
+		api := setupAPI()
+		api.On("KVGet", key).Return([]byte(`{{:}"val-a": 10}`), nil)
+		p.API = api
+
+		var dat map[string]interface{}
+
+		err := p.KVGetJSON(key, &dat)
+
+		api.AssertExpectations(t)
+		assert.NotNil(t, err)
+		assert.Nil(t, dat)
+	})
+
+	t.Run("Valid parameters passed (happy-path)", func(t *testing.T) {
+		key := "test-key"
+
+		p := &plugin.HelpersImpl{}
+
+		api := setupAPI()
+		api.On("KVGet", key).Return([]byte(`{"val-a": 10}`), nil)
+		p.API = api
+
+		var dat map[string]interface{}
+
+		err := p.KVGetJSON(key, &dat)
+
+		api.AssertExpectations(t)
+		assert.Nil(t, err)
+		assert.Equal(t, map[string]interface{}{
+			"val-a": float64(10),
+		}, dat)
+	})
+}
+
+func TestKVSetJSON(t *testing.T) {
+	key := "test-key"
+
+	setupAPI := func() *plugintest.API {
+		return &plugintest.API{}
+	}
+
+	t.Run("JSON Marshal error", func(t *testing.T) {
+		api := setupAPI()
+		api.AssertNotCalled(t, "KVSetJSON")
```

Comment:
Shouldn't this be `KVSet`? 🤔 

```suggestion
		api.AssertNotCalled(t, "KVSet")
```